services:
  sync:
    image: wazoplatform/wait
    environment:
      TIMEOUT: "${INTEGRATION_TEST_TIMEOUT}"

  auth:
    image: wazoplatform/wazo-auth-mock
    ports:
      - "9497"

  rabbitmq:
    image: rabbitmq
    container_name: rabbitmq
    ports:
      - "5672"
    volumes:
      - type: tmpfs
        target: /var/lib/rabbitmq

  provd:
    image: wazo-provd-tests
    volumes:
      - "../..:/usr/src/wazo-provd/"
      - "./etc/wazo-provd/config.yml:/etc/wazo-provd/config.yml"
      - "./keys/wazo-provd-key.yml:/var/lib/wazo-auth-keys/wazo-provd-key.yml"
      - "./jsondb/configs/base:/var/lib/wazo-provd/jsondb/configs/base"
      - "./jsondb/configs/default:/var/lib/wazo-provd/jsondb/configs/default"
      - "./jsondb/configs/defaultconfigdevice:/var/lib/wazo-provd/jsondb/configs/defaultconfigdevice"
      - "./jsondb/configs/autoprov:/var/lib/wazo-provd/jsondb/configs/autoprov"
      # - "${LOCAL_GIT_REPOS}/xivo-lib-python/xivo:/opt/venv/lib/python3.11/site-packages/xivo"
    ports:
      - "8666"
      - "8667"
      - "69/udp"

  pluginserver:
    image: wazo-provd-plugin-server
    volumes:
      - "./plugins/pkgs:/var/www"
    ports:
      - "8080"
